profiles/views.py

```diff
@@ -102,3 +102,26 @@ def destroy(self, request, pk=None):
         profile.save()
         serializer = ProfileSerializer(profile)
         return Response(serializer.data, status=status.HTTP_204_NO_CONTENT)
+
+
+class ViewedCompanyList(ListCreateAPIView):
+    queryset = ViewedCompany.objects.all()
+    serializer_class = ViewedCompanySerializer
+    permission_classes = (IsAuthenticated, )
+
+    def list(self, request, *args, **kwargs):
+        user_id = self.request.user.id
+        viewed_companies = ViewedCompany.objects.filter(user=user_id)
+        serializer = self.serializer_class(viewed_companies, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def create(self, request, *args, **kwargs):
+        user_id = self.request.user.id
+        company_id = request.data.get("profile_id")
+        company = get_object_or_404(Profile, profile_id=company_id)
+        if company.person.pk != user_id:
+            serializer = self.serializer_class(data={"user": user_id, "company": company_id})
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Simple but important rule (please share with the rest of the team).

If you're 'handcrafting' response in DRF code — 99% you're doing something wrong.

You must not do `return Response(...)`. 

In the line 127 you return `400 BAD REQUEST` but there is no data, what exactly is wrong with the request (because by handcrafting request you just break everything which is already implemented in DRF, in a proper way).

Reply:
Now I do not override `create`.